_posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md

```diff
@@ -0,0 +1,254 @@
+a parallel sub-pipeline.
+
+The pipeline configuration for the preceding example is similar to the following:
+
+```
+trace-error-metrics-pipeline:
+  source:
+    pipeline:
+      name: "span-pipeline"
+  processor:
+    - aggregate:
+        identification_keys: ["serviceName", "traceId"]
+        action:
+          count:
+        group_duration: "20s"
+        aggregate_when: "/status_code == 2"
+  sink:
+     - opensearch
+         hosts: ["https://opensearch:9200"]
+        insecure: true
+        username: "admin"
+        password: "admin"
+        index: trace-error-metrics
+        
+ trace-high-latency-metrics-pipeline:
+  source:
+    pipeline:
+      name: "span-pipeline"
+  processor:
+    - aggregate:
+        identification_keys: ["serviceName", "traceId"]
+        action:
+          histogram:
+            key: "durationInNaos"
+            record_minmax: true
+            units: "nanoseconds"
+            buckets: [1000000000, 1500000000, 2000000000]
+        group_duration: "20s"
+        aggregate_when: "/durationInNanos > 1000000000"
+  sink:
+      - opensearch
+         hosts: ["https://opensearch:9200"]
+         insecure: true
+         username: "admin"
+         password: "admin"
+         index: trace-high-latency-metrics
+```
+
+## Anomaly detection
+
+Data Prepper 2.1 introduces anomaly detection as an independent processor that can be placed anywhere in the pipeline to
+detect anomalies using artificial intelligence/machine learning (AI/ML) techniques. For now, the processor only supports the random cut forest algorithm for detecting anomalies. 
+Support for other AI/ML algorithms may be added in the future. The processor can be configured to use any numerical (integer 
+or floating-point data type) field in the input events to detect anomalies.
+
+Anomaly detection in Data Prepper provides a more scalable and efficient way to detect anomalies in logs and traces 
+because the ML algorithm is applied on the server side.
+
+For example, consider an aggregate processor that generates a histogram of trace latencies (`durationInNanos`). If you place an anomaly detector processor after the aggregate processor, it can identify abnormally high latencies. 
+The following is an example of such a configuration:
+
+```
+trace-metric-anomaly-detector-pipeline:
+  source:
+    pipeline:
+      name: "trace-high-latency-metrics-pipeline"
+  processor:
+    - anomaly_detector:
+        keys: ["max"]
+        mode:
+          random_cut_forest:
+  sink:
+    - opensearch
+         hosts: ["https://opensearch:9200"]
+         insecure: true
+         username: "admin"
+         password: "admin"
+         index: high-latency-traces-with-anomalies
+```
+
+## Sampling and rate limiting
+
+Data Prepper 2.1 supports sampling and rate limiting to limit the number of events that are sent to a sink.
+
+These features can be used to reduce the load on the OpenSearch cluster when storing uninteresting logs and metrics.
```

Comment:
```suggestion
These features can be used to reduce the load on the OpenSearch cluster when storing ordinary logs and metrics.
```

Reply:
@natebower , Is there a specific reason you recommended going back to "ordinary" here? It does seem that "uninteresting" is a more apt word because these are logs that somebody would not be very interested in.

Reply:
@dlvenable I think "ordinary" is ok as long as we qualify what we're talking about, which I believe you did in the sentence that follows this one. "Uninteresting" is subjective and a bit awkward in this context.